src/Microsoft.Azure.SignalR.Common/Utilities/ConnectionStringParser.cs

```diff
@@ -97,7 +115,50 @@ internal static (string endpoint, string accessKey, string version, int? port, s
                 }
             }
 
-            return (dict[EndpointProperty].TrimEnd('/'), dict[AccessKeyProperty], version, port, clientEndpoint);
+            AccessKey accessKey;
+            if (dict.ContainsKey(AuthTypeProperty) && "aad".Equals(dict[AuthTypeProperty], StringComparison.OrdinalIgnoreCase))
```

Comment:
use `enum` and `enum.TryParse` for authType

Reply:
How many authTypes do we have, currently in my mind:
```cs
msi,
aad,
keybased,
anonymous, (available for websocket)
```

thoughts?

Reply:
I think `msi` should be part of `aad` since it doesn't make sense beyond AAD scope.